mindmeld/components/entity_resolver.py

```diff
@@ -562,38 +776,829 @@ def _predict_exact_match(self, entity):
+                                            size equal to number of synonyms
+            top_n (int): maximum number of results to populate
+            entity_emb (np.array): a 2d array of embedding of the input entity text(s)
+        Returns:
+            Union[dict, list[tuple]]: if return_as_dict, returns a dictionary of synonyms and their
+                                        scores, else a list of sorted synonym names, paired with
+                                        their similarity scores (descending)
+        """
+
+        n_entities = len(entity_emb)
+
+        is_single = n_entities == 1
+        as_tensor = _get_module_or_attr("torch", "as_tensor")
+        cosine_similarity = _get_module_or_attr("torch", "cosine_similarity")
+
+        # [n_syns, emd_dim] -> [n_entities, n_syns, emd_dim]
+        t_syn_enc = as_tensor(synonyms_encodings)
+        t_syn_enc = t_syn_enc.expand([n_entities, *t_syn_enc.shape])
+
+        # [n_entities, emd_dim] -> [n_entities, n_syns, emd_dim]
+        t_entity_emb = as_tensor(entity_emb)
+        t_entity_emb = t_entity_emb.unsqueeze(dim=1).expand_as(t_syn_enc)
+
+        # returns -> [n_entities, n_syns]
+        similarity_scores_2d = cosine_similarity(t_syn_enc, t_entity_emb, dim=-1).numpy()
+
+        results = []
+        for similarity_scores in similarity_scores_2d:
+            similarity_scores = similarity_scores.reshape(-1)
+            similarity_scores = np.around(similarity_scores, decimals=2)
+
+            if return_as_dict:
+                results.append(dict(zip(synonyms.keys(), similarity_scores)))
             else:
-                self.fit()
+                # results in descending scores
+                n_scores = len(similarity_scores)
+                if n_scores > top_n:
+                    top_inds = similarity_scores.argpartition(n_scores - top_n)[-top_n:]
+                    result = sorted(
+                        zip(np.asarray([*synonyms.keys()])[top_inds], similarity_scores[top_inds]),
+                        key=lambda x: x[1], reverse=True)
+                else:
+                    result = sorted(zip(synonyms.keys(), similarity_scores), key=lambda x: x[1],
+                                    reverse=True)
+                results.append(result)
 
-        except EsConnectionError as e:
-            logger.error(
-                "Unable to connect to Elasticsearch: %s details: %s", e.error, e.info
+        if is_single:
+            return results[0]
+
+        return results
+
+    def _fit(self, clean):
+        """
+        Fits the resolver model
+
+        Args:
+            clean (bool): If ``True``, deletes existing dump of synonym embeddings file
+        """
+
+        cache_path = self.cache_path(
+            app_path=self.app_path,
+            er_config=self.er_config,
+            entity_type=self.type
+        )
+
+        if clean and os.path.exists(cache_path):
```

Comment:
if not clean?

Reply:
we'll not care if the cache exists or not if clean=False. the `os.path.exists(cache_path)` is to just make sure we don't have to handle FileNotFound errors from os library if the cached file didn't exist.